### PR TITLE
feat(provider/gae): Modify deploy description and validators for arti…

### PR DESCRIPTION
**Pull request by jtk54**
…facts.

### CODE REVIEW

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy [hunk 1]
```diff
@@ -46,13 +47,29 @@ class AppengineGcsRepositoryClient implements AppengineRepositoryClient {
...
+    }
+    def slash = fullPath.indexOf("/")
+    def bucketName = fullPath.substring(0, slash)
+    def bucketPath = fullPath.substring(slash + 1)
 
     if (fullPath.endsWith(".tar")) {
-      StorageUtils.untarStreamToPath(storage.openObjectStream(bucketName, bucketPath), dest)
+      InputStream tas = storage.openObjectStream(bucketName, bucketPath)
+
+      // NOTE: We write the tar file out to an intermediate temp file because the tar input stream
```
Comment (lwander) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy]: :|

Comment (jtk54) [reply clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy]: You'd think someone would have solved this before...

File: clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy [hunk 2]
```diff
@@ -46,13 +47,29 @@ class AppengineGcsRepositoryClient implements AppengineRepositoryClient {
...
 
     def dest = targetDirectory + File.separator + applicationDirectoryRoot
-    def fullPath = repositoryUrl.substring(gsPrefix.length()) + '/' + applicationDirectoryRoot
-    def slash = fullPath.indexOf("/");
-    def bucketName = fullPath.substring(0, slash);
-    def bucketPath = fullPath.substring(slash + 1);
+
+    def fullPath = repositoryUrl.substring(gsPrefix.length())
+    if (applicationDirectoryRoot) {
+      fullPath + '/' + applicationDirectoryRoot
```
Comment (duftler) [inline clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy]: This line looks to be missing something.

Comment (jtk54) [reply clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy]: Yeah that will be a problem, fix incoming.

**Pull request merged**